linux/src/socket_transport_linux.c

```diff
@@ -581,32 +581,36 @@ int socket_transport_listen(SOCKET_TRANSPORT_HANDLE socket_transport, uint16_t p
     return result;
 }
 
-SOCKET_TRANSPORT_HANDLE socket_transport_accept(SOCKET_TRANSPORT_HANDLE socket_transport)
+SOCKET_ACCEPT_RESULT socket_transport_accept(SOCKET_TRANSPORT_HANDLE socket_transport, SOCKET_TRANSPORT_HANDLE* accepting_socket)
 {
     SOCKET_TRANSPORT* result;
-    // Codes_SOCKET_TRANSPORT_LINUX_11_069: [ If socket_transport is NULL, socket_transport_accept shall fail and return NULL. ]
+    SOCKET_ACCEPT_RESULT aresult;
+    // Codes_SOCKET_TRANSPORT_LINUX_11_069: [ If socket_transport is NULL, socket_transport_accept shall fail and return SOCKET_ACCEPT_ERROR. ]
     if (socket_transport == NULL)
     {
         LogError("Invalid arguments: SOCKET_TRANSPORT_HANDLE socket_transport: %p", socket_transport);
         result = NULL;
```

Comment:
>result = NULL; [](http://example.com/codeflow?start=8&length=14)

You no longer need to use result because if the function fails you should not modify accepting_socket.  The user must know that if there's a failure then accepting_socket should not be looked at.